src/rcc.rs

```diff
@@ -37,12 +47,36 @@ pub struct Rcc {
 
 #[allow(unused)]
 const HSI: u32 = 8_000_000; // Hz
+#[allow(dead_code)] // Only used for ADC, but no configuration for ADC clock yet.
+const HSI14: u32 = 14_000_000; // Hz - ADC clock.
+// Can't do this anymore... #[cfg(any(feature = "stm32f042", feature = "stm32f072"))]
+const HSI48: u32 = 48_000_000; // Hz - (available on STM32F04x, STM32F07x and STM32F09x devices only)
+
+pub enum SysClkSource {
+    HSI = 0b00,
+    HSE = 0b01,
+    PLL = 0b10,
+    HSI48 = 0b11,
+}
+
+pub enum PllSource {
+    HSI_2 = 0b00,
+    HSI = 0b01,
+    HSE = 0b10,
+    HSI48 = 0b11,
+}
 
 #[allow(unused)]
 pub struct CFGR {
     hclk: Option<u32>,
     pclk: Option<u32>,
     sysclk: Option<u32>,
+    enable_hsi: Option<bool>,
```

Comment:
What is the logic for making all of these booleans options? Given that it looks like later you `unwrap_or` them all later with default values. Why not just make them plain booleans and initialising them with the default values?

Reply:
Good point.  I was trying to be consistent, but maybe that isn't called for here.